ASTGen: account for alternate path separators

compnerd opened the pull request:
Windows supports `\` and `/` as path separators.  Account for both in computing the basename.  This repairs a few test failures on Windows.

compnerd committed:
ASTGen: account for alternate path separators

Windows supports `\` and `/` as path separators.  Account for both in
computing the basename.  This repairs a few test failures on Windows.

compnerd commented:
@swift-ci please test

CodaFi commented on lib/ASTGen/Sources/ASTGen/SourceManager+MacroExpansionContext.swift:
```diff
@@ -43,7 +43,7 @@ extension String {
   /// directory.
   var basename: String {
-    guard let lastSlash = lastIndex(of: "/") else {
+    guard let lastSlash = lastIndex(where: { ["/", "\\"].contains($0) }) else {
```
This condition should be specialized for Windows specifically

compnerd replied:
Avoiding the special casing is nice - is there some case that you expect this to be problematic for on other platforms?

compnerd replied:
Special cased Apple and Linux platforms instead.

compnerd committed:
ASTGen: special case the Apple and Linux platforms

There is no real reason to split the behaviour here as this is a file
system primitive.  Special case Apple and Linux platforms to address
feedback.

compnerd commented:
@swift-ci please test
